src/GitElephant/Command/Caller/CallerSSH2.php

```diff
@@ -52,20 +52,32 @@ public function __construct($resource, $gitPath = '/usr/bin/git')
      * @param string      $cmd the command
      * @param bool        $git prepend git to the command
      * @param null|string $cwd directory where the command should be executed
+     * @param array       $acceptedExitCodes the exit codes to accept.
+     *                      The exit checked is from stream_get_contents:
+     *                      if it returns false, a error is thrown if this array does not contain '1'.
+     *                      For backwards compatibility it does accept false by stream_get_contents by default
      *
      * @return CallerInterface
      */
-    public function execute($cmd, $git = true, $cwd = null): \GitElephant\Command\Caller\CallerInterface
-    {
+    public function execute(
+        $cmd,
+        $git = true,
+        $cwd = null,
+        array $acceptedExitCodes = [0, 1]
```

Comment:
I think checking of returned data should be moved to other places regardingy S in SOLID.
And I would say that streams are not executables, there are no "exit codes" in terms of linux binaries, naming is  confusing, besides that 0 and 1 here are magic numbers..

Reply:
True. I will change this somehow – I guess I have to rethink the compatibility with the CallerInterface first.